From 389ed0e18c10a8b3368d0b9fbd2e1f5a7e37f072 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Wed, 20 May 2020 22:12:52 +0200
Subject: [PATCH] custtomize configure and Makefile to our needs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>

---
 configure | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/configure b/configure
index 4fb2cf0..37f502e 100755
--- a/configure
+++ b/configure
@@ -331,6 +331,7 @@ mingw32-cross)
 	_host_cpu=`echo $guessed_host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\1/'`
 	_host_os=`echo $guessed_host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\3/'`
 	_host_vendor=`echo $guessed_host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\2/'`
+	_host_prefix=$_host
 	;;
 esac
 
@@ -557,8 +558,9 @@ if test -n "$_host"; then
 			_host_os=unix
 			;;
 		*)
-			echo "Cross-compiling to unknown target, please add your target to configure."
-			exit 1
+			echo "Cross-compiling for Yocto Project based systems."
+			DEFINES="$DEFINES -DUNIX"
+			_host_os=unix
 			;;
 	esac
 
@@ -802,7 +804,7 @@ fi
 # Now, add the appropriate defines/libraries/headers
 #
 echo
-find_sdlconfig
+#find_sdlconfig
 
 SRC="src"
 CORE="$SRC/emucore"
@@ -821,7 +823,7 @@ SQLITE="$SRC/common/repository/sqlite"
 
 INCLUDES="-I$CORE -I$COMMON -I$TV -I$TIA -I$TIA_FRAME_MANAGER"
 
-INCLUDES="$INCLUDES `$_sdlconfig --cflags`"
+INCLUDES="$INCLUDES `pkg-config --cflags sdl2`"
 if test "$_build_static" = yes ; then
 	_sdl_conf_libs="--static-libs"
 	LDFLAGS="-static $LDFLAGS"
@@ -829,7 +831,7 @@ else
 	_sdl_conf_libs="--libs"
 fi
 
-LIBS="$LIBS `$_sdlconfig $_sdl_conf_libs`"
+LIBS="$LIBS `pkg-config $_sdl_conf_libs sdl2`"
 LD=$CXX
 
 case $_host_os in
-- 
2.21.1

