From f4fa367ab6e79cbfd77254237335f383311006ec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Sat, 9 Feb 2019 22:16:52 +0100
Subject: [PATCH] Set fixed VICEDIR
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index c5139fd..402e893 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5010,6 +5010,7 @@ case "$host_os" in
     AM_CONDITIONAL(ALTERNATE_DOCDIR, false)
     ;;  
 esac
+VICEDIR="$libdir/vice"
 AC_SUBST(VICEDIR)
 AC_DEFINE_UNQUOTED(VICEDIR, "$VICEDIR",
 		   [Where should we lookup for data files?])
-- 
2.20.1

