From 9a496ab29e585b157f608ea477ae0a55602081ad Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Sat, 22 Oct 2016 00:33:09 +0200
Subject: [PATCH] fix sdl build
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 src/arch/sdl/ssi2001-drv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/arch/sdl/ssi2001-drv.c b/src/arch/sdl/ssi2001-drv.c
index d218be0..e6437d6 100755
--- a/src/arch/sdl/ssi2001-drv.c
+++ b/src/arch/sdl/ssi2001-drv.c
@@ -30,6 +30,7 @@
 
 #ifdef UNIX_COMPILE
 #include "../unix/ssi2001-drv.c"
+#include "../unix/io-access.c"
 #endif
 
 #if defined(WIN32_COMPILE) && !defined(__XBOX__)
-- 
2.5.5

