From 59fbf0c8d72e22caa7e19afa48c031f8822f276e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Wed, 6 Jun 2018 23:38:38 +0200
Subject: [PATCH] Align Makefiles for cross compiling
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>

---
 Makefile      | 39 ++++++---------------------------------
 misc/Makefile | 10 +++++-----
 2 files changed, 11 insertions(+), 38 deletions(-)

diff --git a/Makefile b/Makefile
index 0cda4d6..5b9ead0 100644
--- a/Makefile
+++ b/Makefile
@@ -1,30 +1,4 @@
-# Unix gcc or DOS go32 cross-compiling gcc
-#
-CC = gcc
-LD = gcc
-# for testing. not to be used; build failures in misc/.
-#CFLAGS = -O2 -W -Wall -pedantic -ansi
-CFLAGS = -O2
-LDFLAGS = -lc
-
-# for DOS?
-# CC = gcc-go32
-# LD = gcc-go32
-# CFLAGS = -W -Wall -pedantic 
-
-# Other cc
-#CC = cc
-#CFLAGS =
-#LD = ld
-
-DESTDIR = /usr/local
-
-BINDIR = $(DESTDIR)/bin
-MANDIR = $(DESTDIR)/share/man/man1
-DOCDIR = $(DESTDIR)/share/doc
-
-MKDIR = mkdir -p
-INSTALL = install
+LD = $(CC)
 
 all: killxa xa uncpk
 
@@ -32,7 +6,7 @@ killxa:
 	rm -f xa
 
 xa:
-	(cd src && LD=${LD} CC="${CC} ${CFLAGS}" ${MAKE})
+	(cd src && LD="${LD}" CC="${CC} ${CFLAGS}" ${MAKE})
 
 load:	
 	(cd loader && CC="${CC} ${CFLAGS}" ${MAKE})
@@ -56,11 +30,10 @@ clean:
 	rm -f xa *.exe *.o65
 
 install: xa uncpk
-	$(MKDIR) $(BINDIR)
-	$(MKDIR) $(MANDIR)
-	$(INSTALL) xa reloc65 ldo65 file65 printcbm uncpk $(BINDIR)
-	$(INSTALL) man/file65.1 man/ldo65.1 man/printcbm.1 man/reloc65.1 man/uncpk.1 man/xa.1 $(MANDIR)
-	#$(MKDIR) $(DOCDIR)/xa65
+	install -d $(DESTDIR)$(PREFIX)/bin
+	install -m 755 xa reloc65 ldo65 file65 printcbm uncpk $(DESTDIR)$(PREFIX)/bin
+	install -d $(DESTDIR)$(PREFIX)/share/man/man1
+	install -m 644 man/file65.1 man/ldo65.1 man/printcbm.1 man/reloc65.1 man/uncpk.1 man/xa.1 $(DESTDIR)$(PREFIX)/share/man/man1
 
 dist: clean
 	cd .. ; tar cvf xa-2.3.11.tar xa-2.3.11 ; gzip xa-2.3.11.tar
diff --git a/misc/Makefile b/misc/Makefile
index 2b450e0..6967dad 100644
--- a/misc/Makefile
+++ b/misc/Makefile
@@ -10,19 +10,19 @@ LIBS = #-lncurses -ltermcap -lm
 all: ../mkrom.sh ../uncpk ../printcbm ../file65 ../reloc65 ../ldo65
 
 ../uncpk: uncpk.c
-	${CC} ${CFLAGS} uncpk.c -o $(XCBMLIB)/uncpk
+	${CC} ${CFLAGS} $(LDFLAGS) uncpk.c -o $(XCBMLIB)/uncpk
 
 ../printcbm: printcbm.c
-	${CC} ${CFLAGS} printcbm.c -o $(XCBMLIB)/printcbm
+	${CC} ${CFLAGS} $(LDFLAGS) printcbm.c -o $(XCBMLIB)/printcbm
 
 ../file65: file65.c
-	${CC} ${CFLAGS} file65.c -o $(XCBMLIB)/file65
+	${CC} ${CFLAGS} $(LDFLAGS) file65.c -o $(XCBMLIB)/file65
 
 ../ldo65: ldo65.c
-	${CC} ${CFLAGS} ldo65.c -o $(XCBMLIB)/ldo65
+	${CC} ${CFLAGS} $(LDFLAGS) ldo65.c -o $(XCBMLIB)/ldo65
 
 ../reloc65: reloc65.c
-	${CC} ${CFLAGS} reloc65.c -o $(XCBMLIB)/reloc65
+	${CC} ${CFLAGS} $(LDFLAGS) reloc65.c -o $(XCBMLIB)/reloc65
 
 ../mkrom.sh: mkrom.sh
 	cp mkrom.sh ../mkrom.sh
