From 95788b1b09b0bd8e89cc3c7e5ed769ae3e87ac34 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Sun, 31 Jul 2016 04:02:51 +0200
Subject: [PATCH] fix build with pedantic gcc6
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 src/joystick.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/joystick.c b/src/joystick.c
index 63dd75d..bf99647 100644
--- a/src/joystick.c
+++ b/src/joystick.c
@@ -42,7 +42,6 @@ bool ignore_joystick = false;
 int joysticks = 0;
 Joystick *joystick = NULL;
 
-static const char joystick_cfg_version = 1;
 static const int joystick_analog_max = 32767;
 
 // eliminates axis movement below the threshold
-- 
2.5.5

