From 5b23877f74f9cbfc8a6442e19c672ba91229f923 Mon Sep 17 00:00:00 2001
From: Dan Church <amphetamachine@gmail.com>
Date: Thu, 8 Jul 2021 16:15:37 -0500
Subject: [PATCH] Fix missing include

Upstream-Status: Accepted[https://github.com/dosbox-staging/dosbox-staging/commit/5b23877f74f9cbfc8a6442e19c672ba91229f923]
---
 src/hardware/ne2000.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/hardware/ne2000.cpp b/src/hardware/ne2000.cpp
index 7b4dcec4..3f68fe79 100644
--- a/src/hardware/ne2000.cpp
+++ b/src/hardware/ne2000.cpp
@@ -26,6 +26,7 @@
 
 #if C_NE2000
 
+#include <cstdarg>
 #include <cstdio>
 #include <cstring>
 
-- 
2.31.1

